core/sync/primitives.odin

```diff
@@ -7,10 +7,21 @@ current_thread_id :: proc "contextless" () -> int {
 	return _current_thread_id()
 }
 
-// A Mutex is a mutual exclusion lock
-// The zero value for a Mutex is an unlocked mutex
+// A Mutex is a [mutual exclusion lock](https://en.wikipedia.org/wiki/Mutual_exclusion).
```

Comment:
Odin's docs are not markdown based but use our own custom format. To add a link, it'll need to be in this form:
```
[[mutual exclusion lock; https://en.wikipedia.org/wiki/Mutual_exclusion]]
```